### PR TITLE
Show Previous 3 Months Data For Standup data 

### PR DESCRIPTION
**Pull request by mailmesriza98**
closes https://github.com/Real-Dev-Squad/website-dashboard/issues/428 
Description:
The standup page currently fetches and displays data for the current month only. It uses a table to present the standup data of users, showing their daily progress throughout the current month.
However we show show dates chronologically from current date till 2 months back in a reverse order while skipping Sundays

demo:

https://github.com/Real-Dev-Squad/website-dashboard/assets/22434603/41fc74f4-848b-405f-bb60-7b25606c3917

image

test coverage
testcoverage

### CODE REVIEW

**Comment by Pratiyushkumar** [comment]
Please add some information into PR and also a demo video and also test coverage as well

**Comment by vinit717** [comment]
Please add test coverage report not test passing screenshot

File: standup/script.js [hunk 1]
```diff
@@ -13,9 +13,29 @@ const daysInCurrentMonth = new Date(
...
-  month: 'long',
-});
+
+function isSunday(date) {
+  return date.getDay() === 0; // 0 represents Sunday
+}
+const currentMonthNum = currentDateObj.getMonth();
+
+const endDate = currentDateObj;
+const startDate = new Date(currentYearNum, currentMonthNum - 3, 1); // Start from 3 months ago
```
Comment (vinit717) [inline standup/script.js:23]: Does the requirement for static value? can we make it dynamic?

File: standup/script.js [hunk 2]
```diff
@@ -28,39 +48,75 @@ async function fetchStandupData(userId) {
...
+      let date = new Date(endDate);
+      date >= startDate;
+      date = new Date(date.getTime() - oneDay)
+    ) {
+      if (!isSunday(date)) {
+        standupData.standupFrequency.push('❌');
+        standupData.completedText.push('No data');
+        standupData.plannedText.push('No data');
+        standupData.blockersText.push('No data');
+        //${date.getDate()} ${date.toLocaleString('default', {month: 'long',})} ${date.getFullYear()}
```
Comment (vinit717) [inline standup/script.js:74]: ??

File: standup/script.js [hunk 3]
```diff
@@ -13,9 +13,33 @@ const daysInCurrentMonth = new Date(
...
+  const day = dateObject.getDate();
+  return `${day}-${month}-${year}`;
+}
+
+function formatDate(dateObject) {
+  const year = dateObject.getFullYear();
+  const month = dateObject.getMonth() + 1;
+  const day = dateObject.getDate();
+  return `${day}-${month}-${year}`;
+}
```
Comment (Aryex82) [inline standup/script.js:42]: can you move these utility functions to utils.js

**Comment by Aryex82** [comment]
Can we also add day of the week in table header? Eg: Mon, 14 Aug 2023

File: __tests__/standup/standup.test.js [hunk 4]
```diff
@@ -115,4 +146,14 @@ describe('Standup Page', () => {
...
   });
+
+  it('should display the correct date range in the table header', async () => {
+    const dateCellValues = await page.evaluate(() => {
+      const dateCells = Array.from(document.querySelectorAll('.dates'));
+      return dateCells.map((cell) => cell.textContent.trim());
+    });
+    const expectedDateValues = generateExpectedDateValues();
+    console.log('expected ', expectedDateValues);
+    expect(dateCellValues.length).toEqual(expectedDateValues.length);
```
Comment (Pratiyushkumar) [inline __tests__/standup/standup.test.js:157]: remove console log please

**Review by Pratiyushkumar — COMMENTED** [review]
rest looks good to me..remove console log and I will approve it

**Review by Pratiyushkumar — APPROVED** [review]

**Review by bksh05 — APPROVED** [review]

**Review by sahsisunny — DISMISSED** [review]

**Comment by Aryex82** [comment]
The cell width of each day is not consistent. Can you we have same width for all the columns?

**Review by Aryex82 — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**